backend/src/main/java/fk/prof/backend/http/LeaderHttpVerticle.java

```diff
@@ -181,30 +227,70 @@ private void handleGetWork(RoutingContext context) {
     }
   }
 
-  private void handlePostPolicy(RoutingContext context) {
+  private void handleGetPolicy(RoutingContext context) {
     try {
-      String appId = context.request().getParam("appId");
-      String clusterId = context.request().getParam("clusterId");
-      String procName = context.request().getParam("procName");
-
-      // for now expecting a json payload
-      String payload = context.getBodyAsString("utf-8");
-
-      BackendDTO.RecordingPolicy.Builder builder = BackendDTO.RecordingPolicy.newBuilder();
-      JsonFormat.parser().merge(payload, builder);
-
-      BackendDTO.RecordingPolicy policy = builder.build();
-      Recorder.ProcessGroup pg = Recorder.ProcessGroup.newBuilder().setAppId(appId).setCluster(clusterId).setProcName(procName).build();
+      Recorder.ProcessGroup pG = parseProcessGroup(context);
+      PolicyDTO.VersionedPolicyDetails versionedPolicyDetails = policyStore.getVersionedPolicy(pG);
+      if (versionedPolicyDetails == null) {
+        context.response().setStatusCode(400).end("Policy not found for ProcessGroup " + RecorderProtoUtil.processGroupCompactRepr(pG));
+      } else {
+        context.response().end(ProtoUtil.buildBufferFromProto(versionedPolicyDetails));
+      }
+    } catch (Exception ex) {
+      HttpFailure httpFailure = HttpFailure.failure(ex);
+      HttpHelper.handleFailure(context, httpFailure);
+    }
+  }
 
-      policyStore.put(pg, policy);
-      context.response().end();
+  private void handleCreatePolicy(RoutingContext context) {
+    try {
+      Recorder.ProcessGroup pG = parseProcessGroup(context);
+      PolicyDTO.VersionedPolicyDetails versionedPolicyDetails = parseVersionedPolicyFromPayload(context);
+      policyStore.createVersionedPolicy(pG, versionedPolicyDetails).setHandler(ar -> setResponse(ar, context));
+    } catch (Exception ex) {
+      HttpFailure httpFailure = HttpFailure.failure(ex);
+      HttpHelper.handleFailure(context, httpFailure);
     }
-    catch (Exception ex) {
+  }
+
+  private void handleUpdatePolicy(RoutingContext context) {
+    try {
+      Recorder.ProcessGroup pG = parseProcessGroup(context);
+      PolicyDTO.VersionedPolicyDetails versionedPolicyDetails = parseVersionedPolicyFromPayload(context);
+      policyStore.updateVersionedPolicy(pG, versionedPolicyDetails).setHandler(ar -> setResponse(ar, context));
+    } catch (Exception ex) {
       HttpFailure httpFailure = HttpFailure.failure(ex);
       HttpHelper.handleFailure(context, httpFailure);
     }
   }
 
+  private void setResponse(AsyncResult<? extends AbstractMessage> ar, RoutingContext context){
+    if (ar.succeeded()) {
+      try {
+        context.response().setStatusCode(201).end(ProtoUtil.buildBufferFromProto(ar.result()));
```

Comment:
Discussed with Rohit. Let's do 200 on PUT, 201 on POST as mentioned in mozilla doc